linkerd/protocol/http/src/e2e/scala/io/buoyant/linkerd/protocol/HttpEndToEndTest.scala

```diff
@@ -499,6 +497,40 @@ class HttpEndToEndTest extends FunSuite with Awaits with MustMatchers {
 
   }
 
+  test("timestampHeader adds header") {
+
+    val downstream = Downstream.mk("dog") {
+      req =>
+        req.headerMap.keys must contain ("x-request-start")
+        val rsp = Response()
+        rsp.status = Status.Ok
+        rsp
+    }
+
+    val yaml =
+      s"""|routers:
+          |- protocol: http
+          |  dtab: /svc/* => /$$/inet/127.1/${downstream.port}
+          |  servers:
+          |  - port: 0
+          |    timestampHeader: x-request-start
+          |""".stripMargin
+    val linker = Linker.load(yaml)
+    val router = linker.routers.head.initialize()
+    val s = router.servers.head.serve()
+
+    val req = Request()
+    req.host = "test"
+
+    val c = upstream(s)
+    try {
+      val resp = await(c(req))
```

Comment:
to ensure that this request was even routed to the downstream correctly, should assert the status code is ok